packages/dcos-launch/extra/launch/config.py

```diff
@@ -1,16 +1,38 @@
+import atexit
 import copy
 import os
 
 import yaml
 
-import dcos_installer.config
 import gen
 import launch.util
 import ssh.validate
 import test_util.aws
 from gen.internals import resolve_configuration, Scope, Source, Target, validate_one_of
 from pkgpanda.util import load_string, load_yaml, YamlParseError
 
+# dcos_installer.config will directly import gen.build_deploy.util
+# which expects to find the image commit in the environment or in
+# a directory-local git tree **at import time**. Therefore, the
+# environment variable DCOS_IMAGE_COMMIT must be set here.
+if 'DCOS_IMAGE_COMMIT' not in os.environ:
+    os.environ['DCOS_IMAGE_COMMIT'] = ''
+    atexit.register(os.unsetenv, 'DCOS_IMAGE_COMMIT')
+
+import dcos_installer.config  # noqa
```

Comment:
Too much work for this PR. That function can also parse the current git head instead of reading from the environment, so that means that it relies on current working directory which means anytime pkpanda uses it will be prone to breaking if called at runtime. I strongly feel trying to tackle that in this PR is out of scope.